propertyestimator/layers/reweighting.py

```diff
@@ -42,24 +90,188 @@ def schedule_calculation(calculation_backend, storage_backend, layer_directory,
+            `existing_data`, which have been serialized with `serialize_force_field`
+        """
+        property_class = physical_property.__class__
+
+        if not hasattr(property_class, 'reweight'):
+            return None
+
+        existing_data = {}
+        existing_force_fields = {}
+
+        for substance_id in existing_data_paths:
+
+            existing_data[substance_id] = []
+
+            for data_path in existing_data_paths[substance_id]:
+
+                with open(data_path, 'rb') as file:
+                    existing_data[substance_id].append(pickle.load(file))
+
+        for force_field_id in existing_force_field_paths:
+
+            with open(existing_force_field_paths[force_field_id], 'rb') as file:
+                existing_force_fields[force_field_id] = deserialize_force_field(pickle.load(file))
+
+        reweighted_property = property_class.reweight(physical_property, options, parameter_set_id, existing_data,
+                                                      existing_force_fields, available_resources)
+
+        return_object = CalculationLayerResult()
+        return_object.property_id = physical_property.id
+
+        return_object.calculated_property = reweighted_property
+        return_object.data_to_store = []
+
+        return return_object
+
+    @staticmethod
+    def get_reduced_potential(substance, thermodynamic_state, target_force_field_id,
+                              target_force_field, existing_data, available_resources):
+        """Get the reduced potential from an existing data set. If the target force field
+        id does not match the force field id used by the existing data, the reduced potential
+        will be re-evaluated using the target force field.
+
+        Parameters
+        ----------
+        thermodynamic_state: The thermodynamic state to use when reducing the
+            energies.
+        target_force_field_id: str
+            The id of the target force field parameters.
+        target_force_field: ForceField
+            The target set of force field parameters.
+        existing_data: StoredSimulationData
+            The reference data to get the energies from.
+
+        Returns
+        -------
+        :obj:`numpy.ndarray`
+            The reduced potential of the target state.
+        """
+
+        # If the parameters haven't changed, we only need to swap out the T / P?
+        potential_energies = existing_data.statistics_data.get_observable(ObservableType.PotentialEnergy)
+
+        if target_force_field_id != existing_data.parameter_set_id:
+
+            potential_energies = ReweightingLayer.resample_data(substance, existing_data.trajectory_data,
+                                                                target_force_field, available_resources)
+
+        reduced_potentials = []
+
+        for index, potential_energy in enumerate(potential_energies):
+            reduced_potentials.append(potential_energy / unit.kilojoule_per_mole)
```

Comment:
We should never silently use a specific arbitrary unit like this. There are only two ways to store dimensionless quantities, and we need to be clear about documenting them:
* The [reduced potential](http://www.alchemistry.org/wiki/Multistate_Bennett_Acceptance_Ratio#Reduced_potential) is the dimensionless ratio `potential_energy / kT`, not the value in kJ/mol. It is recommended we use this definition for potential energies whenever reweighting or statistical mechanics is performed with the energies, since the probability density is just `p(x) \propto \exp[-u(x)]`, which makes this simple to work with.
* For all other cases---including when estimating average enthalpies or energies using observables---if we need to store dimensionless quantities, we should use `quantity.value_in_unit_system(unit_system)` where we standardize the unit system to be, for example, [SI units](https://github.com/pandegroup/openmm/blob/master/wrappers/python/simtk/unit/unit_definitions.py#L308-L316).

Reply:
Good catch - I'm not sure why this was added as it redundant in this case anyway!